src/main/java/io/numaproj/numaflow/shared/GrpcServerHelper.java

```diff
@@ -0,0 +1,110 @@
+package io.numaproj.numaflow.shared;
+
+import io.grpc.BindableService;
+import io.grpc.Context;
+import io.grpc.Contexts;
+import io.grpc.ForwardingServerCallListener;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.grpc.netty.NettyServerBuilder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.unix.DomainSocketAddress;
+
+import static io.numaproj.numaflow.shared.GrpcServerUtils.DATUM_METADATA_WIN_END;
+import static io.numaproj.numaflow.shared.GrpcServerUtils.DATUM_METADATA_WIN_START;
+import static io.numaproj.numaflow.shared.GrpcServerUtils.WINDOW_END_TIME;
+import static io.numaproj.numaflow.shared.GrpcServerUtils.WINDOW_START_TIME;
+
+public class GrpcServerHelper {
+    private EventLoopGroup bossEventLoopGroup;
+    private EventLoopGroup workerEventLoopGroup;
+
+    public void gracefullyShutdownEventLoopGroups() {
+        if (this.bossEventLoopGroup != null) {
+            this.bossEventLoopGroup.shutdownGracefully();
+        }
+        if (this.workerEventLoopGroup != null) {
+            this.workerEventLoopGroup.shutdownGracefully();
+        }
+    }
+
+    public Server createServer(
```

Comment:
This is mostly copied from GrpcServerUtils.java. We need to manage the state of event loop groups hence I created this class.